File: Talleres/LaboRobot_2/BattleStrategy.java
package aspitia.mattei;

import robocode.JuniorRobot;

public interface BattleStrategy {
    /**
     * run: entrypoint
     */
    void run(JuniorRobot robot);

    /**
     * onScannedRobot: What to do when you see another robot
     */
    void onScannedRobot(JuniorRobot robot);

    /**
     * onHitByBullet: What to do when you're hit by a bullet
     */
    void onHitByBullet(JuniorRobot robot);

    /**
     * onHitWall: What to do when you hit a wall
     */
    void onHitWall(JuniorRobot robot);

    /**
     * onHitRobot: What to do when you hit a robot
     */
    void onHitRobot(JuniorRobot robot);
}

File: Talleres/LaboRobotKotlin_1/BattleStrategy.kt
package aspitia.mattei.kotlin

import robocode.JuniorRobot


interface BattleStrategy {
    /**
     * run: entrypoint
     */
    fun run(robot: JuniorRobot)

    /**
     * onScannedRobot: What to do when you see another robot
     */
    fun onScannedRobot(robot: JuniorRobot)

    /**
     * onHitByBullet: What to do when you're hit by a bullet
     */
    fun onHitByBullet(robot: JuniorRobot)

    /**
     * onHitWall: What to do when you hit a wall
     */
    fun onHitWall(robot: JuniorRobot)

    /**
     * onHitRobot: What to do when you hit a robot
     */
    fun onHitRobot(robot: JuniorRobot)
}